sklearn/tests/test_common.py

```diff
@@ -68,36 +90,37 @@ def _tested_estimators():
         yield name, estimator
 
 
-def _generate_checks_per_estimator(check_generator, estimators):
-    with ignore_warnings(category=(DeprecationWarning, FutureWarning)):
-        for name, estimator in estimators:
-            for check in check_generator(name, estimator):
-                yield estimator, check
-
-
-def _rename_partial(val):
-    if isinstance(val, functools.partial):
-        kwstring = "".join(["{}={}".format(k, v)
-                            for k, v in val.keywords.items()])
-        return "{}({})".format(val.func.__name__, kwstring)
-    # FIXME once we have short reprs we can use them here!
-    if hasattr(val, "get_params") and not isinstance(val, type):
-        return type(val).__name__
-
-
 @pytest.mark.parametrize(
         "estimator, check",
-        _generate_checks_per_estimator(_yield_all_checks,
-                                       _tested_estimators()),
-        ids=_rename_partial
-)
+        chain.from_iterable(check_estimator(estimator, generate_only=True)
+                            for _, estimator in _tested_estimators()),
+        ids=set_check_estimator_ids)
 def test_estimators(estimator, check):
     # Common tests for estimator instances
     with ignore_warnings(category=(DeprecationWarning, ConvergenceWarning,
                                    UserWarning, FutureWarning)):
         set_checking_parameters(estimator)
-        name = estimator.__class__.__name__
-        check(name, estimator)
+        check(estimator)
+
+
+def test_check_estimator_generate_only():
+    estimator_cls_gen_checks = check_estimator(LogisticRegression,
+                                               generate_only=True)
+    all_instance_gen_checks = check_estimator(LogisticRegression(),
+                                              generate_only=True)
+    assert isgenerator(estimator_cls_gen_checks)
+    assert isgenerator(all_instance_gen_checks)
+
+    estimator_cls_checks = list(estimator_cls_gen_checks)
+    all_instance_checks = list(all_instance_gen_checks)
+
+    # all classes checks include check_parameters_default_constructible
+    assert len(estimator_cls_checks) == len(all_instance_checks) + 1
+
+    # Grid search can not be constructed
+    grid_estimator_gen_checks = list(check_estimator(GridSearchCV,
```

Comment:
It would good to add more context why it cannot be constructed (e.g. we don't support support meta-estimators etc.) and if that is expected to change in the future (with a TODO if so).